### PR TITLE
feat: new assert `notPattern`

### PR DESCRIPTION
**Pull request by RomanHotsiy**
## What/Why/How?

Negative regular expressions are really hard to author and maintain.
A negative regexp matcher simplifies a lot of real-life rules.

## Testing

Covered with tests.

## Check yourself

- [x] Code is linted
- [ ] Tested with redoc/reference-docs/workflows
- [x] All new/updated code is covered with tests

## Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### CODE REVIEW

File: __tests__/lint/assertions-pattern-error/snapshot.js [hunk 1]
```diff
@@ -5,7 +5,21 @@ exports[`E2E lint assertions-pattern-error 1`] = `
 validating /openapi.yaml...
 [1] openapi.yaml:20:16 at #/paths/~1pet~1findByStatus/get/summary
 
-Operation summary should have 'test' word
+summary-pattern assertion failed because the Operation summary didn't meet the assertions: "summary example" should match a regex /test/
```
Comment (RomanHotsiy) [inline __tests__/lint/assertions-pattern-error/snapshot.js:8]: I'm not sure why it doesn't combine two assertion messages. Must something unrelated to this PR.

File: docs/rules/custom-rules.md [hunk 2]
```diff
@@ -381,6 +382,20 @@ rules:
       pattern: /test/
 ```
 
+### `notPattern` example
+
+The following example asserts that the operation summary doesn't start with "The".
+
+```yaml
+rules:
+  assert/operation-summary-contains-test:
```
Comment (adamaltman) [inline docs/rules/custom-rules.md:391]:
```suggestion
  assert/no-operation-summary-starts-with-the:
```

File: docs/rules/custom-rules.md [hunk 3]
```diff
@@ -381,6 +382,20 @@ rules:
...
+    subject:
+      type: Operation
+      property: The summary
```
Comment (adamaltman) [inline docs/rules/custom-rules.md:394]:
```suggestion
      property: summary
```

File: __tests__/lint/assertions-pattern-error/snapshot.js (hunk 1 above)
Comment (IgorKarpiuk) [reply __tests__/lint/assertions-pattern-error/snapshot.js:8]: @RomanHotsiy We will create a separate PR to fix it

**Pull request merged**